src/expr.rs

```diff
@@ -674,9 +674,13 @@ impl<'a> Rewrite for Loop<'a> {
             ControlBraceStyle::AlwaysNextLine => alt_block_sep.as_str(),
             ControlBraceStyle::AlwaysSameLine => " ",
         };
+        let block_width = try_opt!(width.checked_sub(label_string.len() + self.keyword.len() +
+                                                     extra_offset(&pat_expr_string, inner_offset) +
+                                                     1));
```

Comment:
OK, sweet, thanks for the explanation. Could you add a comment here summarising that please? And also on the Rewrite impl for Block.